app/models/lighthouse526_document_upload.rb

```diff
@@ -40,7 +42,7 @@ class Lighthouse526DocumentUpload < ApplicationRecord
     end
 
     event :fail do
-      transitions from: :pending, to: :failed, guard: %i[end_time_saved? error_message_saved?]
+      transitions from: :pending, to: :failed, guard: %i[end_time_saved? error_message_saved?], after: :handle_failure
```

Comment:
This is a nifty callback!

Reply:
Yeah, that AASM magic always enjoyed working with that gem!